src/main/java/cinema/model/ShoppingCart.java

```diff
@@ -0,0 +1,55 @@
+package cinema.model;
+
+import java.util.List;
+import javax.persistence.CascadeType;
+import javax.persistence.Entity;
+import javax.persistence.FetchType;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.OneToMany;
+import javax.persistence.OneToOne;
+
+@Entity
+public class ShoppingCart {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @OneToMany(cascade = CascadeType.ALL, fetch = FetchType.EAGER)
```

Comment:
let's not use fetch type Eager for OneToMany relationships. And you are using join fetch type in you query anywat

Reply:
Why CascadeType.All?